src/Servers/Kestrel/Core/src/Http2Limits.cs

```diff
@@ -141,5 +145,55 @@ public int InitialStreamWindowSize
                 _initialStreamWindowSize = value;
             }
         }
+
+        /// <summary>
+        /// Gets or sets the keep alive ping interval. The server will send a keep alive ping to the client if it
+        /// doesn't receive any frames for this period of time. This property is used together with
+        /// <see cref="KeepAlivePingTimeout"/> to close broken connections.
+        /// <para>
+        /// Interval must be greater than or equal to 1 second. Set to <see cref="TimeSpan.MaxValue"/> to
+        /// disable the keep alive ping interval.
+        /// Defaults to <see cref="TimeSpan.MaxValue"/>.
+        /// </para>
+        /// </summary>
+        public TimeSpan KeepAlivePingInterval
+        {
+            get => _keepAlivePingInterval;
+            set
+            {
+                // Keep alive uses Kestrel's system clock which has a 1 second resolution. Time is greater or equal to clock resolution.
+                if (value < Heartbeat.Interval && value != Timeout.InfiniteTimeSpan)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), CoreStrings.FormatArgumentTimeSpanGreaterOrEqual(Heartbeat.Interval));
+                }
+
+                _keepAlivePingInterval = value != Timeout.InfiniteTimeSpan ? value : TimeSpan.MaxValue;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the keep alive ping timeout. Keep alive pings are sent when a period of inactivity exceeds
+        /// the configured <see cref="KeepAlivePingInterval"/> value. The server will close the connection if it
+        /// doesn't receive any frames within the timeout.
+        /// <para>
+        /// Timeout must be greater than or equal to 1 second. Set to <see cref="TimeSpan.MaxValue"/> to
```

Comment:
IMO we don't need to mention `Time.InfiniteTimeSpan`. The other TimeSpan properties don't mention it, and really it is just an internal implementation detail that if you happen to set InfiniteTimeSpan to any of the TimeSpan properties on the limits class that it will automatically change it to TimeSpan.MaxValue for you.

Reply:
If it's not worth documenting then it's not worth supporting.

Reply:
I agree, but this is only for consistency with the other TimeSpan properties in this area.

Reply:
I'm perfectly fine with how this was done.

It might have been nicer if we started with everything as `TimeSpan?`, but I think consistency is the most important thing here. I'm happy we still support `Time.InfiniteTimeSpan` like we do for other APIs, but we don't need to document it. Recommending MaxValue in docs seems fine to me since that works everywhere.